packages/cache/src/providers/BackfillCacheProvider.ts

```diff
@@ -0,0 +1,127 @@
+import { createBackfillCacheConfig, createBackfillLogger } from "../backfillWrapper";
+import { getCacheStorageProvider } from "backfill-cache";
+import { getPackageInfos } from "workspace-tools";
+import { promisify } from "util";
+import * as fs from "fs";
+import * as path from "path";
+import type { CacheProvider, CacheProviderOptions } from "../types/CacheProvider";
+import type { Logger as BackfillLogger } from "backfill-logger";
+import type { PackageInfo } from "workspace-tools";
+import type { Target } from "@lage-run/target-graph";
+
+const rmdir = promisify(fs.rmdir);
+const rm = promisify(fs.unlink);
+const readdir = promisify(fs.readdir);
+const stat = promisify(fs.stat);
+
+const MS_IN_A_DAY = 1000 * 60 * 60 * 24;
+
+export interface BackfillCacheProviderOptions {
+  root: string;
+  cacheOptions: Partial<CacheProviderOptions>;
+  isReadOnly?: boolean;
+}
+
+export class BackfillCacheProvider implements CacheProvider {
```

Comment:
Move definitions to their own files, so that the index can just be about what gets exported.